service/common/src/main/java/org/apache/polaris/service/admin/PolarisAdminService.java

```diff
@@ -474,8 +474,10 @@ private void authorizeGrantOnTableLikeOperationOrThrow(
       if (status.getFailedToResolvePath().getLastEntityType() == PolarisEntityType.TABLE_LIKE) {
         if (subType == PolarisEntitySubType.ICEBERG_TABLE) {
           throw new NoSuchTableException("Table does not exist: %s", identifier);
-        } else {
+        } else if (subType == PolarisEntitySubType.ICEBERG_VIEW) {
           throw new NoSuchViewException("View does not exist: %s", identifier);
+        } else if (subType == PolarisEntitySubType.GENERIC_TABLE) {
```

Comment:
can just put it as `else` or if there can any other value, lets throw IllegalStateException ?